plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/GroovySourceFileAllowlist.java

```diff
@@ -152,6 +148,27 @@ private static boolean endsWithIgnoreCase(String value, String suffix) {
         }
     }
 
+    @Extension
+    @SuppressFBWarnings(value = "NP_BOOLEAN_RETURN_NULL", justification = "intentionally not caching negative results")
+    public static class AllowedGroovyResourcesCache {
```

Comment:
This could just be folded into `GroovySourceFileAllowlist`; there is no real reason to have a separate extension object for it.

Reply:
Yes, but then it would have to be a static field, and it holds state, which I think we prefer to avoid in static fields.

Reply:
Oh, I missed that `GroovySourceFileAllowlist` is `abstract` and not itself an `@Extension`. Never mind.